src/components/common/Comment.js

```diff
@@ -1,29 +1,105 @@
-import React, { useState } from "react";
-import "../../styles/index.css";
+import React, { useState, useEffect } from "react";
+import { Button } from "antd";
+import Parse from "parse/dist/parse.min.js";
 
-function Comment() {
-  // Initialize state variables using the useState hook
+function Comment({ locationId }) {
   const [comments, setComments] = useState([]); // Store the list of comments
   const [newComment, setNewComment] = useState(""); // Store the new comment being typed by the user
+  const [hashtags, setHashtags] = useState([]); // Keeping track of available hashtags
+  const [selectedHashtags, setSelectedHashtags] = useState([]); // Keeping track of selected hashtags
+  const [checkedStatus, setCheckedStatus] = useState({});
+
+  useEffect(() => {
+    const fetchHashtags = async () => {
+      const Hashtags = Parse.Object.extend("Hashtags");
+      const query = new Parse.Query(Hashtags);
+
+      try {
+        const hashtagResults = await query.find();
+        setHashtags(hashtagResults);
+      } catch (error) {
+        console.error(`Error while fetching Hashtags: ${error}`);
+      }
+    };
+
+    // Call the fetch function
+    fetchHashtags();
+  }, []);
 
   // Function to add a new comment to the list
-  const addComment = () => {
+  const addComment = async () => {
+    // Fetch data of the current user
+    const currentUser = Parse.User.current();
+
     if (newComment) {
-      setComments([...comments, newComment]); // Update the list of comments with the new comment
-      setNewComment(""); // Clear the input field
+      try {
+        // Create Parse object for the Comment class and save it to the database
+        const Comment = Parse.Object.extend("Comment");
+        const commentToSave = new Comment();
+
+        // Set class attributes
+        commentToSave.set("CommentText", newComment);
+        commentToSave.set("UserID", currentUser);
```

Comment:
We call it UserID because in our database, we have a pointer in the comment class to the User class, which then fetches the UserID (which is the name of the column in the comment class). When changing it to currentUser.id, parse complains as it expects a user Object, and can not take a string.